src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -375,6 +376,23 @@ public PaginatedResult getProjects(final Tag tag) {
         return getProjects(tag, false, false, false);
     }
 
+    /**
+     * Returns an optional containing the project version with the latest BOM import date.
+     * @param name the name of the project
+     * @return the version of the project with the latest BOM update date
+     */
+    @Override
+    public Optional getLastImportedVersionProject(final String name) {
+        final PaginatedResult result = getProjects(name, false, false, null);
+        return result.getObjects()
+                .stream()
+                .min((o1, o2) -> {
+                    Project p1 = (Project) o1;
+                    Project p2 = (Project) o2;
+                    return p2.getLastBomImport().compareTo(p1.getLastBomImport());
+                });
```

Comment:
The same outcome could be achieved in a single database query instead. In JDOQL, it would be (roughly, haven't tested):

```java
Query<Project> query = pm.newQuery(Project.class);
query.setFilter("name == :name");
query.setParameters(name);
query.setOrdering("lastBomImport DESC");
query.setRange(0, 1);
return Optional.ofNullable(query.executeUnique());
```